libs/application/ui-shell/src/lib/ApplicationForm.tsx

```diff
@@ -24,12 +26,17 @@ const ApplicationLoader: FC<{
   applicationId: string
   nationalRegistryId: string
 }> = ({ applicationId, nationalRegistryId }) => {
-  const { data, error, loading } = useQuery(GET_APPLICATION, {
+  const { data, error, loading, refetch } = useQuery(GET_APPLICATION, {
     variables: {
       input: {
         id: applicationId,
       },
     },
+    // Setting this so that refetch causes a re-render
+    // https://github.com/apollographql/react-apollo/issues/321#issuecomment-599087392
+    // We want to refetch after setting the application back to 'draft', so that
+    // it loads the correct form for the 'draft' state.
+    notifyOnNetworkStatusChange: true,
```

Comment:
When using `notifyOnNetworkStatusChange`, you will also get a change to the loading boolean going from false, to true, to false again. How it's done at the moment, we gonna render a blank page until the loading pass to false again. Maybe we could use this [loading state](https://ui.devland.is/?path=/docs/navigation-loadingicon--template) instead.

Reply:
Good catch, updated it to use the loading icon